program-runtime/src/loaded_programs.rs

```diff
@@ -795,6 +795,16 @@ impl<FG: ForkGraph> LoadedPrograms<FG> {
         working_slot: &S,
         keys: impl Iterator<Item = (Pubkey, (LoadedProgramMatchCriteria, u64))>,
     ) -> Arc<Mutex<ExtractedPrograms>> {
+        debug_assert!(self.fork_graph.is_some());
+        debug_assert_eq!(
+            Some(working_slot.current_epoch()),
+            self.fork_graph
+                .as_ref()
+                .unwrap()
+                .read()
+                .unwrap()
+                .slot_epoch(working_slot.current_slot())
+        );
```

Comment:
I've added the asserts to ensure no other tests are created without `BankForks`. Please, @pgarg66 let me know if I should remove them.

Reply:
We can remove the `debug_assert_eq!`. Looks like with this PR all the tests are fixed up. In the next PR you can remove the `WorkingSlot`, and just pass the current_slot as `Slot` value.

Reply:
Should I keep the first `debuf_assert`?
The second one ensures `BankForks` is properly set on the tests, because we fetch the epoch from it.

Reply:
I've remove the `debug_assert_eq`.